app/models/miq_region.rb

```diff
@@ -136,10 +136,14 @@ def self.destroy_region(conn, region, tables = nil)
   end
 
   def self.remote_replication_type?
+    return false unless PglogicalSubscription.logical_replication_supported?
+
     MiqPglogical.new.provider?
```

Comment:
interesting how all of these method calls seem to redirect to `MiqPglogical.new`

at least all of the ones that you are guarding.

ugh. looks like they do not use a factory after all.